src/LibraryManager/Providers/jsDelivr/JsDelivrCatalog.cs

```diff
@@ -0,0 +1,286 @@
+                    completionSet.Length = version.Length;
+
+                    IEnumerable<string> versions;
+
+                    if (IsGitHub(name).Result)
+                    {
+                        versions = await GetGithubLibraryVersions(name);
+                    }
+                    else
+                    {
+                        JsDelivrLibraryGroup libGroup = new JsDelivrLibraryGroup(name);
+                        versions = await libGroup.GetLibraryIdsAsync(CancellationToken.None);
+                    }
+
+                    foreach (string libraryId in versions)
+                    {
+                        CompletionItem completionItem = new CompletionItem
+                        {
+                            DisplayText = libraryId,
+                            InsertionText = libraryId
+                        };
+
+                        completions.Add(completionItem);
+                    }
+                }
+
+                completionSet.Completions = completions;
+            }
+            catch (Exception ex)
+            {
+                _provider.HostInteraction.Logger.Log(ex.ToString(), LogLevel.Error);
+            }
+
+            return completionSet;
+        }
+
+        public async Task<IReadOnlyList<ILibraryGroup>> SearchAsync(string term, int maxHits, CancellationToken cancellationToken)
+        {
+            List<ILibraryGroup> libraryGroups = new List<ILibraryGroup>();
+
+            if (IsGitHub(term).Result)
+            {
+                return libraryGroups;
+            }
+
+            try
+            {
+                IEnumerable<string> packageNames = await Microsoft.Web.LibraryManager.Providers.Unpkg.NpmPackageSearch.GetPackageNamesAsync(term, CancellationToken.None);
+                libraryGroups = packageNames.Select(packageName => new JsDelivrLibraryGroup(packageName)).ToList<ILibraryGroup>();
+            }
+            catch (Exception ex)
+            {
+                _provider.HostInteraction.Logger.Log(ex.ToString(), LogLevel.Error);
+            }
+
+            return libraryGroups;
+        }
+
+        public async Task<IEnumerable<string>> GetGithubLibraryVersions(string name)
+        {
+            List<string> versions = new List<string>();
+            JObject versionsObject = await WebRequestHandler.Instance.GetJsonObjectViaGetAsync(string.Format(LatestLibraryVersionUrlGH, name), CancellationToken.None).ConfigureAwait(false);
+            JArray versionsArray = versionsObject["versions"] as JArray;
+
+            foreach (string version in versionsArray)
+            {
+                versions.Add(name + "@" + version);
+            }
+
+            return versions;
+        }
+
+        public static async Task<bool> IsGitHub(string libraryId)
+        {
+            if (libraryId == null)
+            {
+                return false;
+            }
+
+            return libraryId.Contains('/');
```

Comment:
This check would probably be incorrect for scoped npm packages like `@angular/cli`

https://cdn.jsdelivr.net/npm/@angular/cli@1.0.0/

Reply:
It is but scoped packages are not supported by the current Npm/Unpkg/LibraryNaming implementations anyway so we thought support can be added later here as well.